### PR TITLE
GCS_MAVLink: warn on FTP transfer request when FTP disabled

### PR DESCRIPTION
**Pull request by kd0aij**

### CODE REVIEW

**Comment by WickedShell** [comment]
Why do we need this at all? This should be covered by the capability bits, and a GCS should be managing this behind the scenes directly for the user.

If we really wanted something here we should be sending an actual NACK message, not a status text. (We also need to implement the target_system/target_component stuff to filter down on a shared MAVLink network (we probably should do this anyway though)).

Finally to do a statustext I think you'd need a rate limiter, or a GCS attempting to list files will just spam you with statustexts.

**Comment by kd0aij** [comment]
If clearing the capability bit is sufficient then we can just remove the check in ftp_init 
But can we rely on the GCS respecting the capabilities bits?
If not then perhaps this should just fail silently as before (without calling ftp_init)

**Comment by WickedShell** [comment]
A good GCS should have those before asking for FTP. The GCS implementations that are out there? Hard to say honestly. To the best of my knowledge most systems will ask for the capabilities bits once on connect and stash them, so if we change it later that won't really help them.

We could send an actual NACK back here if we wanted (an FTP NACK I mean, not a repurposing of command_int or similiar)

**Comment by kd0aij** [comment]
There's also the chance that ftp_init will fail to allocate buffers when a second GPS requests an FTP transfer, right?
Would sending a NACK in that case be the right thing to do?

**Comment by kd0aij** [comment]
Since I don't know enough about mavlinkftp to answer these questions, I'll just tag this for the devcall

File: libraries/GCS_MAVLink/GCS_Common.cpp [hunk 1]
```diff
@@ -3285,7 +3285,12 @@ void GCS_MAVLINK::handle_common_message(const mavlink_message_t &msg)
         break;
 
     case MAVLINK_MSG_ID_FILE_TRANSFER_PROTOCOL:
-        handle_file_transfer_protocol(msg);
+        // check if ftp is disabled for memory savings
+        if (AP_BoardConfig::ftp_disabled()) {
+            gcs().send_text(MAV_SEVERITY::MAV_SEVERITY_ALERT, "MAVFTP not supported");
```
Comment (tridge) [inline libraries/GCS_MAVLink/GCS_Common.cpp:3290]: we should send a ftp NACK not a text msg

**Review by tridge — CHANGES_REQUESTED** [review]

**Comment by WickedShell** [comment]
The core thing this is trying to help with is if FTP is disabled, which if that was the case then the capability bit was never set. Any GCS who requests FTP while it's been explicitly disabled is a GCS bug, so we don't need to do anything in this case. We should do the FTP nack if we can't allocate the buffers, but that's a seperate PR then this, so I'm going to close this for the moment, as I don't think there is anything to really change in what this is currently doing. Feel free to reopen if you disagree.

**Pull request closed without merging**